### PR TITLE
Refactor `fs-transformer` to enable v4 version of the test

### PR DESCRIPTION
**Pull request by will-cromar**

### CODE REVIEW

**Comment by will-cromar** [comment]
I made the `PyTorchTpuVmMixin` more like the JAX one, where we can substitute in extra setup steps as needed without having to change the model run command. I also moved the epic stack of runtime flags into a list to clean up the messy string manipulation. This change will make it easier to update settings for the v4 version of the test.

Checked the diff. Most changes are either whitespace or rearranging flags. This should be a no-op for existing tests.

File: templates/tpus.libsonnet [hunk 1]
```diff
@@ -22,6 +22,7 @@ local base = import 'base.libsonnet';
     type: 'tpu',
     version: error 'Must override `version`',
     size: error 'Must override `size`',
+    numCores: if tpu.version <= 3 then 8 else 4,
     replicas: tpu.size / 8,  // Each TPU replica has 8 cores
```
Comment (allenwang28) [inline templates/tpus.libsonnet:26]: do we also need to change this?

`replicas: tpu.size / self.numCores`

**Review by allenwang28 — CHANGES_REQUESTED** [review]
Thanks for cleaning this up! This should be a great reference for others

File: templates/tpus.libsonnet (hunk 1 above)
Comment (will-cromar) [reply templates/tpus.libsonnet:26]: This calculation should still be correct -- `replicas` essentially means hosts in this context.

**Review by allenwang28 — APPROVED** [review]

**Pull request merged**